backend/tournesol/lib/public_dataset.py

```diff
@@ -170,3 +177,86 @@ def get_individual_criteria_scores_data(poll_name: str) -> QuerySet:
         """,
         {"poll_name": poll_name},
     )
+
+
+def write_comparisons_file(poll_name: str, write_target) -> None:
+    """
+    Write the output of `get_comparisons_data` as CSV in `write_target`, an
+    object supporting the Python file API.
+    """
+
+    # If we want this function to be generic, the specific video_a and video_b
+    # columns should be renamed entity_a and entity_b.
+    fieldnames = [
+        "public_username",
+        "video_a",
+        "video_b",
+        "criteria",
+        "score",
+    ]
+    writer = csv.DictWriter(write_target, fieldnames=fieldnames)
+    writer.writeheader()
+    writer.writerows(
+        {
+            "public_username": comparison.username,
+            "video_a": comparison.uid_a.split(UID_DELIMITER)[1],
+            "video_b": comparison.uid_b.split(UID_DELIMITER)[1],
+            "criteria": comparison.criteria,
+            "score": int(round(comparison.score)),
+        }
+        for comparison in get_comparisons_data(poll_name).iterator()
+    )
+
+
+def write_users_file(poll_name: str, write_target) -> None:
+    """
+    Write the output of `get_users_data` as CSV in `write_target`, an object
+    supporting the Python file API.
+    """
+    fieldnames = [
+        "public_username",
+        "trust_score",
+    ]
+    writer = csv.DictWriter(write_target, fieldnames=fieldnames)
+    writer.writeheader()
+    writer.writerows(
+        {
+            "public_username": user.username,
+            "trust_score": user.trust_score,
+        }
+        for user in get_users_data(poll_name).iterator()
+    )
+
+
+def write_individual_criteria_scores_file(poll_name: str, write_target) -> None:
+    """
+    Write the output of `get_individual_criteria_scores_data` as CSV in
+    `write_target`, an object supporting the Python file API.
+    """
+
+    # If we want this function to be generic, the specific video column should
+    # be renamed entity.
+    fieldnames = [
+        "public_username",
+        "video",
+        "criteria",
+        "score",
+        "voting_right",
+    ]
+
+    criteria_scores = get_individual_criteria_scores_data(poll_name).iterator()
+
+    rows = (
+        {
+            "public_username": criteria_score.username,
+            "video": criteria_score.uid.split(UID_DELIMITER)[1],
+            "criteria": criteria_score.criteria,
+            "score": criteria_score.score,
+            "voting_right": criteria_score.voting_right,
```

Comment:
Maybe we could round those two column, it as a lot of decimals

```suggestion
            "score": round(criteria_score.score,1),
            "voting_right": round(criteria_score.voting_right,3),
```

Reply:
Sure. I'm in favor of doing it in another PR to avoid mixing the refactor of the lib and the refactor of the dataset creation. We may need to adapt some tests after these changes.